src/protobuf.zig

```diff
@@ -963,6 +977,91 @@ pub fn pb_decode(comptime T: type, input: []const u8, allocator: Allocator) !T {
     return result;
 }
 
+fn freeAllocated(allocator: Allocator, token: json.Token) void {
+    // Took from std.json source code since it was non-public one
+    switch (token) {
+        .allocated_number, .allocated_string => |slice| {
+            allocator.free(slice);
+        },
+        else => {},
+    }
+}
+
+fn fillDefaultStructValues(
+    comptime T: type,
+    r: *T,
+    fields_seen: *[@typeInfo(T).Struct.fields.len]bool,
+) !void {
+    // Took from std.json source code since it was non-public one
+    inline for (@typeInfo(T).Struct.fields, 0..) |field, i| {
+        if (!fields_seen[i]) {
+            if (field.default_value) |default_ptr| {
+                const default = @as(
+                    *align(1) const field.type,
+                    @ptrCast(default_ptr),
+                ).*;
+                @field(r, field.name) = default;
+            } else {
+                return error.MissingField;
+            }
+        }
+    }
+}
+
+fn parseStructField(
+    comptime T: type,
+    result: *T,
+    comptime fieldInfo: StructField,
+    allocator: Allocator,
+    source: anytype,
+    options: json.ParseOptions,
+) !void {
+    // Is this a good way to check if it's ArrayList or not? Took from here:
+    // https://ziggit.dev/t/how-to-json-de-ser-struct-with-arraylist-field/3510/5
```

Comment:
Suggestion: Instead of trying to deduce it from the name of the type, `fieldInfo` could be of type `FieldDescriptor`. Then you'll just have to check if `ftype` is of type `.List` or `.PackedList`.

Reply:
also need to handle oneOf here.

Reply:
That was a really good advice! Made it.